lib/widgets/main_screen_bottomnavbar.dart

```diff
@@ -0,0 +1,132 @@
+import 'package:provider/provider.dart';
+
+import '../change_notifiers/tab_manager.dart';
+
+class MainScreenBottomNavBar extends StatelessWidget {
+  final Function onTap;
+
+  const MainScreenBottomNavBar({super.key, required this.onTap});
+
+  @override
+  Widget build(BuildContext context) {
+    const double borderRadius = 30.5;
+
+    return Container(
+      decoration: BoxDecoration(
+        color: Theme.of(context).colorScheme.tertiary,
+        borderRadius: const BorderRadius.only(
+          topLeft: Radius.circular(borderRadius),
+          topRight: Radius.circular(borderRadius),
+        ),
+      ),
+      height: 70,
+      padding: const EdgeInsets.only(
+        top: 10,
+        left: 10,
+        right: 10,
+        bottom: 0,
+      ),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+        children: [
+          _SelectedIcon(
+            image: 'assets/custom_svg_icons/home.svg',
+            index: 0,
+            label: 'Home',
+            onTap: onTap,
+          ),
+          _SelectedIcon(
+            image: 'assets/custom_svg_icons/message.svg',
+            index: 1,
+            label: 'Messages',
+            onTap: onTap,
+          ),
+          _SelectedIcon(
+            image: 'assets/custom_svg_icons/profile.svg',
+            index: 2,
+            label: 'Profile',
+            onTap: onTap,
+          ),
+          _SelectedIcon(
+            image: 'assets/custom_svg_icons/settings.svg',
+            index: 3,
+            label: 'Settings',
+            onTap: onTap,
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _SelectedIcon extends StatelessWidget {
+  const _SelectedIcon({
+    required this.image,
+    required this.index,
+    required this.label,
+    required this.onTap,
+    Key? key,
+  }) : super(key: key);
+
+  final String image;
+  final int index;
+  final String label;
+  final Function onTap;
+
+  @override
+  Widget build(BuildContext context) {
+    const double defaultIconHeight = 20;
+    const double selectedIconHeight = 20;
+
+    return Consumer<TabManager>(
+      builder: (context, tabManager, child) {
+        bool isSelected = tabManager.selectedTab == index;
+
+        return TextButton(
+          style: TextButton.styleFrom(
+            padding: const EdgeInsets.all(0),
+            splashFactory: NoSplash.splashFactory,
+          ),
+          onPressed: () {
+            tabManager.goToTab(index);
+            onTap();
+          },
+          child: Column(
+            mainAxisAlignment:
+                (isSelected) ? MainAxisAlignment.end : MainAxisAlignment.start,
+            children: [
+              const SizedBox(
+                height: 10,
+              ),
+              SvgPicture.asset(
+                image,
+                height: (isSelected) ? selectedIconHeight : defaultIconHeight,
+                color: (isSelected)
+                    ? Theme.of(context).colorScheme.secondary
```

Comment:
same comment for theming

Reply:
noted sir. but if you can see i already have an app theme file in lib (lib/app_theme.dart). there are inconsistent elements within the ui which demands different font sizes under the same fontStyle hence i used .copyWith function with those text styles which already exists and having the same weight and color.

Reply:
yeah. seems correct. except for one thing. it doesn't need to be in a class. so your app_theme.dart file should consist of these only.

```dart
final lightTheme = ThemeData(
   // etc.
);

final darkTheme = ThemeData(
   // etc.
);
```

Reply:
okay sir. i only used it in a class so i can name it like this:

AppTheme.lightTheme
AppTheme.darkTheme

i can modify it otherwise if it is not needed. thanks sir!